dutch-auction/src/main.sw

```diff
@@ -0,0 +1,123 @@
+contract;
+
+use std::{
+    address::Address,
+    contract_id::ContractId,
+    block::height,
+    chain::auth::msg_sender,
+    assert::assert,
+    revert::revert,
+    storage::StorageMap,
+    token::transfer_to_output,
+    context::{msg_amount, call_frames::msg_asset_id},
+    result::*,
+    identity::Identity,
+};
+
+abi MyContract {
+    fn get_price() -> u64;
+    fn set_beneficiary(new_beneficiary: Address);
+    fn bid();
+}
+
+//Set this to your own address
+const MY_ADDRESS: b256 = 0x0000000000000000000000000000000000000000000000000000000000000000;
+
+storage {
+    startingPrice: u64,
+    endingPrice: u64,
+    startTime: u64,
+    endTime: u64,
+    beneficiary: Address = ~Address::from(MY_ADDRESS),
+    admin: Address = ~Address::from(MY_ADDRESS),
+    ended: bool,
+    //Enter the asset id of the currency you want the bid to be in
+    asset_id: b256 = 0x0000000000000000000000000000000000000000000000000000000000000000,
+}
+
+fn win() {
+    // Do stuff on the win event
+    transfer_to_output(price(), ~ContractId::from(storage.asset_id), storage.beneficiary);
+    storage.ended = true;
+}
+
+impl MyContract for Contract {
+    fn get_price() -> u64 {
+        return price();
```

Comment:
rename to `_get_price()` and the `return` can be implicit so remove the `return` and the `;` at the end

Reply:
Why the name change?

Reply:
From Slack discussion: no `_` prefix. Implicit return is 👍 though

Reply:
`price()` does not tell me anything about what the function is doing. Is it setting the price in storage, is it returning something etc. Given the change of use for `_` renaming to `get_price()` might cause a compilation error. Nevertheless, based on the logic of the function I think it should still be renamed to something like `calculate_price()` since that is what the function seems to be doing. It calculates it and returns it. The API call (`get_price()`) is fine but I would move that towards the bottom since it's just a getter

Reply:
Renamed it to calculate_price